### DISCUSSION TITLE
How to update users organisation?

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Lern/Community#18] [Query] [answered]
I have a requirement in the admin. Can I update the user organization? Let's say the admin created a user in the root Org 'X' and now he wants to add this user into the sub-organization. Basically, we are trying to develop a user management feature in the project. Admin can do all these activities like add user, block/unblock, etc. Similarly, I need this facility like updating the user's organization. Based on the API and the input this is not possible to implement I guess. 

```
'/learner/user/v2/update': {
      checksNeeded: ['ROLE_CHECK', 'OWNER_CHECK'],
      ROLE_CHECK: [ROLE.ALL],
      OWNER_CHECK: {
        checks: [
          { entity: '__session__userId', params: [] }
        ]
      }
    },
```

@amit-tarento can you please help me here?

**Comment by amitec006** [comment, marked as answer]
@pranavirabatti You can use user upload api to update user organisation as an admin.

**Reply by pranavirabatti** [reply]
I think this is not possible @amit-tarento. The person(admin) who is changing the organization of a user has to have the same org Id. Otherwise, I may end up with this

```
[
    {
        "firstName": "Prakash",
        "phone": null,
        "roles": [
            "CONTENT_CREATOR",
            "CONTENT_REVIEWER"
        ],
        "userType": "teacher",
        "userId": "35e2b633-325c-49f8-b0fd-3bf4b946646c",
        "email": null,
        "orgId": "0135636614830161921",
        "organisationId": "0135636614830161921",
        "operation": "create",
        "err_msg": "Root organisation ID of API user is conflicting with that of specified organisation ID."
    }
] 
```

**Reply by amitec006** [reply]
Admin should belong to same root org as the user , then he can update user sub-org

**Reply by pranavirabatti** [reply]
I did the same. The user 'Prakash' belongs to the same org to which admin was added. Now the admin is trying to update the user with sub-org. It does not work.

**Reply by amitec006** [reply]
and the org which you are updating also belongs to same root org , where admin belongs too. 
Can you confirm on this.

**Reply by pranavirabatti** [reply]
Yes, both org share the same channel.  That sub org which I'm trying to update is created under the admin user root org only.

**Reply by amitec006** [reply]
Then it should work.
Try with this sample csv
`name,phone,email,userType,userId,roles,orgId`
`ww,,w@yopmail.com,teacher,50d5b4e7-2d07-4175-ac90-242acffdc241,"PUBLIC",01310491696352460828627`

where orgId is the new sub-orgid.
And user-auth token should be the admin auth token from same user root org.

**Reply by pranavirabatti** [reply]
We are trying to update it from the admin only. So, I think I may not be having any issues with user-auth token. The sample file is attached below.
[sample_.csv](https://github.com/Sunbird-Lern/Community/files/9018650/sample_.csv)

**Reply by amitec006** [reply]
@pranavirabatti Can you please provide the admin and the uploaded user read response here.

**Reply by pranavirabatti** [reply]
Admin - https://pastebin.com/6TY2YP8Z
User to update - https://pastebin.com/FWApA81J

**Reply by pranavirabatti** [reply]
Amit any update on this?

**Reply by amitec006** [reply]
@pranavirabatti Actually this user upload can update the user org data (but due to code issue its not working, will fix that).
If you are using latest release of learner i.e from release-4.9.0 onwards then you can also use `PATCH   /private/user/v1/update` APi which does not require any user auth token to update user organisation , but this API call should be done via service to service call(its not onboarded)

**Reply by pranavirabatti** [reply]
Ok, fine Amit. We are not allowed to upgrade the service now. I just want to confirm that updating the user's organization is not possible in release-4.5.0. Is that correct?

**Reply by amitec006** [reply]
yes for admin users. user himself can update.

**Reply by pranavirabatti** [reply]
By chance is there any way to add these old APIs? If yes, what code changes do we need to do? 

Member add and delete api: 
http://docs.sunbird.org/3.8.0/developer-docs/how-to-guide/how_to_create_org_add_user/

**Reply by amitec006** [reply]
We removed these APIs in release-3.8.0, But you can refer release-3.7.0 code and update the respective file accordingly to bring back these APIs.

**Reply by pranavirabatti** [reply]
Thank you so much for your help.

**Reply by pranavirabatti** [reply]
Can you please provide the PR where these two APIs were removed?

**Reply by amitec006** [reply]
@pranavirabatti Please refer the below PR 
`https://github.com/Sunbird-Lern/sunbird-lms-service/pull/731`

**Reply by pranavirabatti** [reply]
Ok. Thanks!

**Comment by Govind3877** [comment]
how to delete user or remove EmailId/Phone Number which is link to user?
